### ISSUE TITLE
Idea for exercise (linear regression)

### ISSUE THREAD

**Issue by gimseng** [gimseng/99-ML-Learning-Projects#1]
Goal: Learn how to use linear regression

Packages needed: sklearn.linear_model.LinearRegression

Idea and task: 
- Obtain interesting dataset with rather linear relationship. E.g. include GDP vs Happiness Index, crop yield vs rain fall and etc

- Apply Linear Regression.
- Bonus and extra credits : Understanding outliers, multi-dimensional regressions

**Comment by Ar-Kareem** [comment]
If we were to walk the reader into understanding linear regression from scratch I think these would be the logical steps from introducing a task to explain a small part of it

    1- explain what the task is... input/output (features vs predicted variable)
    2- "How will we solve this?"... present the first ML idea, Linear Regression line formula (y = w*x+b) or (y = b+w_1*x_1+w_2*x_2+...w_n*x_n)
    3- "How to evaluate a good 'w' value?"... present the cost function
    4- "How are we going to pick a good w value" present gradient descent on the cost function

As for what data we could use, there's the classic predicting house prices task ([like this](https://www.kaggle.com/c/house-prices-advanced-regression-techniques)), or we could get some other dataset.

Thoughts?

**Comment by gimseng** [comment]
@Ar-Kareem Thats a great set up ! I totally agree with your approach. Please go ahead and set out the exercise and solution. I cleaned up the contributing guideline a little, bringing some format to how things should be organised.

Please proceed in anyway you see fit, and maybe we can have more discussion when pull request is made. Thanks !

**Comment by Dhruv-VINT** [comment]
Hello @gimseng ! I'm willing to make a detailed Linear Regression model/exercise (for this issue) as part of my first contribution towards open source. Can I reopen this issue and add a new Linear Regression exercise in this??
